### PR TITLE
feat: add Container Messages tab to dev panel

### PR DESCRIPTION
**Pull request by andy-haynes**
This PR adds a tab to the dev panel to display details on messages sent between the outer window application and individual containers. Styling leaves something to be desired but I think it's useful in its current state.

image

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -29,6 +29,8 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-hook-form": "^7.46.2",
+    "react-json-tree": "^0.18.0",
+    "react-json-view": "^1.21.3",
```
Comment (mpeterdev) [inline apps/web/package.json:33]: is `react-json-view` needed or was this an alternative you tested?

File: apps/web/src/components/Messaging.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  containerMessages,
+}: {
+  containerMessages: BWEMessage[];
+}) {
+  const [selectedMessage, setSelectedMessage] = useState<object | null>(null);
+
+  const reversedMessages = [...containerMessages];
+  reversedMessages.reverse();
+
+  console.log({ selectedMessage });
```
Comment (mpeterdev) [inline apps/web/src/components/Messaging.tsx:164]: intentionally left in?

**Comment by mpeterdev** [comment]
great addition 🎉  , just two nits

File: apps/web/src/components/Messaging.tsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      const { requestId, result } = message;
+      return {
+        message,
+        isFromComponent,
+        name: 'return',
+        componentId: parseComponentId(
+          isFromComponent ? message.containerId : message.targetId
+        )!,
+        summary: `[${requestId.split('-')[0]}] returned ${result} ${
+          !isFromComponent ? 'to' : ''
```
Comment (pavelisnear) [inline apps/web/src/components/Messaging.tsx:125]: I'm not sure if this stands against the convention, but might be a bit easier to read?
```suggestion
          isFromComponent ? '' : 'to'
```

File: apps/web/package.json (hunk 1 above)
Comment (andy-haynes) [reply apps/web/package.json:33]: Nice catch! It was the latter.

File: apps/web/src/components/Messaging.tsx (hunk 2 above)
Comment (andy-haynes) [reply apps/web/src/components/Messaging.tsx:164]: Nope, just a very old branch I didn't vet very thoroughly 😅

**Pull request merged**